common/sagas/schedule/currentTimeBounty.ts

```diff
@@ -0,0 +1,35 @@
+import { call, put, takeLatest, select } from 'redux-saga/effects';
+import { SagaIterator } from 'redux-saga';
+import { TypeKeys } from 'actions/schedule/constants';
+import { getDecimal, getUnit } from 'selectors/transaction';
+import { validDecimal, validNumber } from 'libs/validators';
+import { toTokenBase } from 'libs/units';
+import { validateInput } from 'sagas/transaction/validationHelpers';
+import BN from 'bn.js';
+import { SetCurrentTimeBountyAction } from 'actions/schedule/actionTypes/timeBounty';
+import { SetTimeBountyFieldAction, setTimeBountyField } from 'actions/schedule';
+
+export function* setCurrentTimeBounty({ payload: raw }: SetCurrentTimeBountyAction): SagaIterator {
+  const decimal: number = yield select(getDecimal);
+  const unit: string = yield select(getUnit);
+
+  if (!validNumber(parseInt(raw, 10)) || !validDecimal(raw, decimal)) {
+    yield call(setField, { raw, value: null });
+  }
+
+  const value = toTokenBase(raw, decimal);
+  const isInputValid: boolean = yield call(validateInput, value, unit);
+
+  const isValid = isInputValid && value.gte(new BN(0));
+
+  yield call(setField, { raw, value: isValid ? value : null });
+}
+
+export function* setField(payload: SetTimeBountyFieldAction['payload']) {
```

Comment:
This function isn't used anywhere externally, would rather just have inline `yield put(setTimeBountyField(payload))` in the two references above to avoid extra indirection